src/lib/bookmarks/bookmarkstoolbarbutton.cpp

```diff
@@ -61,9 +61,24 @@ void BookmarksToolbarButton::setShowOnlyIcon(bool show)
     updateGeometry();
 }
 
+bool BookmarksToolbarButton::showOnlyText() const
+{
+    return m_showOnlyText;
+}
+
+void BookmarksToolbarButton::setShowOnlyText(bool show)
+{
+    m_showOnlyText = show;
+    updateGeometry();
```

Comment:
Unfortunately, leftmost button in bookmarks panel keeps on showing icon until it is repainted (via hovering). So QPushButton::update() call may be solution for that.